scipy/stats/_distn_infrastructure.py

```diff
@@ -2238,9 +2314,14 @@ def _penalized_nnlf(self, theta, x):
         loc, scale, args = self._unpack_loc_scale(theta)
         if not self._argcheck(*args) or scale <= 0:
             return inf
-        x = asarray((x-loc) / scale)
-        n_log_scale = len(x) * log(scale)
-        return self._nnlf_and_penalty(x, args) + n_log_scale
+        if isinstance(x, CensoredData):
+            x = CensoredData((x._lower - loc)/scale, (x._upper - loc)/scale)
+            n = np.count_nonzero(x._not_censored)
+            return self._censored_nnlf_and_penalty(x, args) + n * log(scale)
```

Comment:
I can see why this makes sense. 
In the nlf, every uncensored term should include the PDF divided by scale, so negative log adds `n*log(scale)`. The censored terms include the CDF, which doesn't need to be divided by scale, so we don't include those terms in `n`.
```python3
from scipy.stats import norm
x = np.random.rand(10)
loc, scale = np.randomom.rand(2)
dist1 = norm(loc=loc, scale=scale)
dist2 = norm()
z = (x - loc) / scale
np.testing.assert_allclose(dist1.pdf(x), dist2.pdf(z)/scale)
np.testing.assert_allclose(dist1.cdf(x), dist2.cdf(z))
```
Might be worth a note for those a little less familiar with the code.